sycl/source/detail/platform_impl.cpp

```diff
@@ -203,6 +203,21 @@ std::vector<platform> platform_impl::get_platforms() {
   // may be initialized after.
   GlobalHandler::registerDefaultContextReleaseHandler();
 
+  // Some applications/libraries prefer to implement their own device selection
+  // and default to just providing the first available device. Make sure that
+  // the first platform has the most preferrable device.
+  auto GetPlatformScore = [](const platform &p) {
+    auto devs = p.get_devices();
+    auto it =
+        std::max_element(devs.begin(), devs.end(), [](auto lhs, auto rhs) {
+          return default_selector_v(lhs) < default_selector_v(rhs);
+        });
+    return default_selector_v(*it);
+  };
+  std::stable_sort(Platforms.begin(), Platforms.end(), [&](auto lhs, auto rhs) {
+    return GetPlatformScore(lhs) > GetPlatformScore(rhs);
+  });
```

Comment:
Using this, I worry we'll have to call `GetPlatformScore` multiple times per platform. Instead, should we calculate them once, put the results in a vector of `std::tuple<int, platform>` and sort by the first element?